hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -79,6 +79,7 @@
       this.inputStream = fsDataInputStream;
     }
 
+    fsDataInputStream.seek(0);
```

Comment:
What is the original exception were you seeing ? 

This seek() call is immediately after open() so should be innocuous for other file-systems. right ?

It would be better to inline document the reasoning behind this change for future understanding.

Reply:
magicBuffer check was failing on file open, as it couldn't find a beginning of the HUDI block in the log file. 
That was throwing exception that was killing compaction. In the debug, it appeared that content of magicBuffer was incorrect, and steam offsets we off. Only happened when more that 1 files was scheduled to be processed. So, I didn't test all the variations originally (non static magicBuffer without seek and seek with static) as was in hurry to fix, so not sure which one actually fixes the issue. Agree that seek(0) is weird and probably non static is a fix. Added comment and additional check..